diff -up at-3.1.10/parsetime.y.fix at-3.1.10/parsetime.y
--- at-3.1.10/parsetime.y.fix	2005-08-05 05:31:04.000000000 +0200
+++ at-3.1.10/parsetime.y	2012-01-02 12:53:08.320823391 +0100
@@ -486,14 +486,16 @@ parsetime(int argc, char **argv)
     if (yyparse() == 0) {
 	if (time_only)
 	{
-	    if ((exectm.tm_hour < currtm.tm_hour) ||
-		((exectm.tm_hour == currtm.tm_hour &&
+	    if (exectm.tm_mday == currtm.tm_mday &&
+		(exectm.tm_hour < currtm.tm_hour ||
+		(exectm.tm_hour == currtm.tm_hour &&
 		    exectm.tm_min <= currtm.tm_min)))
 		exectm.tm_mday++;
 	} 
 	else if (!yearspec) {
-	    if ((exectm.tm_mon < currtm.tm_mon) ||
-	        ((exectm.tm_mon == currtm.tm_mon &&
+	    if (exectm.tm_year == currtm.tm_year &&
+		(exectm.tm_mon < currtm.tm_mon ||
+	        (exectm.tm_mon == currtm.tm_mon &&
 		     exectm.tm_mday < currtm.tm_mday)))
 		exectm.tm_year++;
 	}
