diff -up at-3.1.10/atd.c.crash at-3.1.10/atd.c
--- at-3.1.10/atd.c.crash	2013-09-09 16:45:53.000000000 +0200
+++ at-3.1.10/atd.c	2013-09-09 17:11:45.302560537 +0200
@@ -341,18 +341,15 @@ run_file(const char *filename, uid_t uid
     newname[0] = '=';
 
     /* We try to make a hard link to lock the file.  If we fail, then
-     * somebody else has already locked it (a second atd?); log the
+     * somebody else has already locked or deleted it; log the
      * fact and return.
      */
     if (link(filename, newname) == -1) {
-	if (errno == EEXIST) {
-	    free(mailname);
-	    free(newname);
-	    syslog(LOG_WARNING, "trying to execute job %.100s twice",filename);
-	    return;
-	} else {
-	    perr("Can't link execution file");
-	}
+        syslog(LOG_WARNING,(errno == EEXIST ? "trying to execute job %.100s twice"
+            : "Can't link execution file %.100s: %m"), filename);
+        free(mailname);
+        free(newname);
+        return;
     }
     /* If something goes wrong between here and the unlink() call,
      * the job gets restarted as soon as the "=" entry is cleared
