diff -up at-3.1.10/at.c.fclose at-3.1.10/at.c
--- at-3.1.10/at.c.fclose	2014-11-05 13:04:58.205013748 +0100
+++ at-3.1.10/at.c	2014-12-11 10:41:07.068255792 +0100
@@ -213,7 +213,11 @@ nextjob()
     jobno = (1 + jobno) % 0xfffff;	/* 2^20 jobs enough? */
     fprintf(fid, "%05lx\n", jobno);
 
-    fclose(fid);
+    if (ferror(fid))
+	jobno = EOF;
+
+    if (fclose(fid) != 0)
+	jobno = EOF;
     return jobno;
 }
 
@@ -502,7 +506,8 @@ writefile(time_t runtimer, char queue)
     if (ferror(stdin))
 	panic("Input error");
 
-    fclose(fp);
+    if (fclose(fp) != 0)
+	panic("Output error");
 
     /* Set the x bit so that we're ready to start executing
      */
