diff -up at-3.1.10/at.1.in.help at-3.1.10/at.1.in
--- at-3.1.10/at.1.in.help	2014-12-11 10:43:27.000000000 +0100
+++ at-3.1.10/at.1.in	2015-02-16 15:23:18.204671716 +0100
@@ -9,7 +9,7 @@ at, batch, atq, atrm \- queue, examine o
 .IR queue ]
 .RB [ -f
 .IR file ]
-.RB [ -mldbv ]
+.RB [ -mMlbv ]
 .B TIME
 .br
 .B at
@@ -18,7 +18,7 @@ at, batch, atq, atrm \- queue, examine o
 .IR queue ]
 .RB [ -f
 .IR file ]
-.RB [ -mldbv ]
+.RB [ -mMlbv ]
 .RB -t
 .IR time_arg
 .br
@@ -26,6 +26,10 @@ at, batch, atq, atrm \- queue, examine o
 .I job
 .RI [ job... ]
 .br
+.B "at [ -rd ]"
+.I job
+.RI [ job... ]
+.br
 .B atq
 .RB [ -V ]
 .RB [ -q
@@ -216,6 +220,9 @@ is given a specific queue, it will only
 Send mail to the user when the job has completed even if there was no
 output.
 .TP 8
+.B \-M
+Never send mail to the user.
+.TP 8
 .BI \-f " file"
 Reads the job from
 .BI file
@@ -225,6 +232,10 @@ rather than standard input.
 Is an alias for
 .B atq.
 .TP
+.B \-r
+Is an alias for
+.B atrm.
+.TP
 .B \-d
 Is an alias for
 .B atrm.
diff -up at-3.1.10/panic.c.help at-3.1.10/panic.c
--- at-3.1.10/panic.c.help	2006-01-03 08:01:37.000000000 +0100
+++ at-3.1.10/panic.c	2015-02-16 15:15:00.030197468 +0100
@@ -91,9 +91,11 @@ usage(void)
 {
 /* Print usage and exit.
  */
-    fprintf(stderr, "Usage: at [-V] [-q x] [-f file] [-mldbv] time\n"
+    fprintf(stderr, "Usage: at [-V] [-q x] [-f file] [-mMlbv] timespec ...\n"
+            "       at [-V] [-q x] [-f file] [-mMlbv] -t time\n"
     	    "       at -c job ...\n"
 	    "       atq [-V] [-q x]\n"
+	    "       at [ -rd ] job ...\n"
 	    "       atrm [-V] job ...\n"
 	    "       batch\n");
     exit(EXIT_FAILURE);
